leetcode/medium/227._Basic_Calculator_II.py

```diff
@@ -0,0 +1,51 @@
+# Link to problem: https://leetcode.com/problems/basic-calculator-ii/description/
+# Loom explaining solution you wrote (5m): 
+# Solution:
+# Description
+# Key ideas
+# To step process, process */ first, +- last.
+# - can be treated as + (-val), +- can be mapped as sum
+# When processing */, use a value stack
+# Difficulty: Medium
+# Code
+
+
+def token_of(s):
+    cur = ""
+    for c in s:
+        if c == " ":
+            if cur:
+                yield int(cur)
+                cur = ""
+        elif c in "+-*/":
```

Comment:
Instead of using a strnig like this, you should use a list of a set. 

```
c in ["+", "-", "/", "*"]:
``

Reply:
you mean a list or a set right?

Reply:
You could use either a list or a set. But using a `str` data type is the worst of the 3 options. A list is fine for small lists `O(1)`

Reply:
The ideal case is to either use Enums or define them in advance. 

```
PLUS_OP = "+"
MINUS_OP = "-"
MULTIPLY_OP = "*"
DIVIDE_OP = "/"

ALL_OPS = [PLUS_OP, MINUS_OP, MULTIPLY_OP, DIVIDE_OP]
```

Then you use these constants throughout.

Reply:
> You could use either a list or a set. But using a `str` data type is the worst of the 3 options. A list is fine for small lists `O(1)`

isn't `+-*/` a lot more readable and easier to write? Especially given we can see the search space is small here.

we are not using any specific facility by str/ array, isn't str and array the same here?

What is the expectation for non O(n) impacting "performance" practice during interview? Isn't this the classic "don't measure before you make any performance claim". [e.g. Pure string is literally as fast as a set, while list is the slowest](https://perfpy.com/910)